### PR TITLE
[service_discovery][jmx] enabling for JMX-based integrations (GRPC)

### PR DESCRIPTION
**Pull request by truthbk**
### What does this PR do?

This PR introduces an RPC client to enable the submission of JMX configurations (in YAML format) to JMXfetch, thus enabling service discovery for JMX-based integrations (jmxfetch). The changes introduce a dependency with [grpc.io](http://www.grpc.io/). The PR might require some polishing, but is functional at this time.
### Motivation

There was a fairly obvious void in the functionalities for service discovery with regard to JMX-based integrations. This PR attempts to address that issue.
### Testing Guidelines

Manual, at the moment.
### Additional Notes

Note that the RPC client code is generate at build time in omnibus, so this may cause some issues in terms of linting, etc... due to unfound modules.

https://github.com/DataDog/omnibus-software/pull/79
https://github.com/DataDog/dd-agent-omnibus/pull/96
https://github.com/DataDog/jmxfetch/pull/109
https://github.com/DataDog/docker-dd-agent/pull/135

### CODE REVIEW

File: agent.py [hunk 1]
```diff
@@ -19,6 +19,8 @@
 import signal
 import sys
 import time
+import supervisor.xmlrpc
```
Comment (remh) [inline agent.py]: nitpick but that's 3rd party

File: datadog.conf.example [hunk 2]
```diff
@@ -123,6 +123,9 @@ gce_updated_hostname: yes
 # and modify its value.
 # sd_template_dir: /datadog/check_configs
 #
+# JMX Service Disocvery
```
Comment (remh) [inline datadog.conf.example]: Do we need this flag ? Let's add some comments about why it's needed or not

File: jmxfetch.py [hunk 3]
```diff
@@ -93,6 +97,10 @@ def _handle_sigterm(self, signum, frame):
         log.debug("Caught sigterm. Stopping subprocess.")
         self.jmx_process.terminate()
 
+    def _handle_sigreload(self, signum, frame):
+        # Terminate jmx process on SIGTERM signal
```
Comment (remh) [inline jmxfetch.py]: Why do we need a new signal handler if we are using gRPC ?

File: utils/service_discovery/abstract_config_store.py [hunk 4]
```diff
@@ -275,7 +275,7 @@ def crawl_config_template(self):
         # Initialize the config index reference
         if self.previous_config_index is None:
             self.previous_config_index = config_index
-            return False
+            return True
```
Comment (remh) [inline utils/service_discovery/abstract_config_store.py]: What's the reason for this change ?

File: util.py [hunk 5]
```diff
@@ -124,6 +124,37 @@ def check_yaml(conf_path):
...
+def config_to_yaml(config):
+    '''
+    Convert a config dict to YAML
+    '''
+    assert 'init_config' in config, "No 'init_config' section found"
+    assert 'instances' in config, "No 'instances' section found"
+
+    valid_instances = True
+    if config['instances'] is None or not isinstance(config['instances'], list):
+        valid_instances = False
```
Comment (remh) [inline util.py]: You might as well raise here.

**Comment by hkaj** [comment]
@truthbk can we close it?

**Comment by truthbk** [comment]
@hkaj yes we can. Sorry about that.
